src/compiler/program.ts

```diff
@@ -1635,6 +1637,7 @@ namespace ts {
 
             sourceFileToPackageName = oldProgram.sourceFileToPackageName;
             redirectTargetsMap = oldProgram.redirectTargetsMap;
+            usesUriStyleNodeCoreModules = oldProgram.usesUriStyleNodeCoreModules;

```

Comment:
Does this get fixed up on subsequent builds?

Reply:
Yeah, you got it. It gets reused if resolution isn’t going to happen in the next program.

Reply:
There’s a test that shows editing back and forth: https://github.com/microsoft/TypeScript/pull/45080/files#diff-0694f66b155b0b58bf9818822b2ff472fb62409216c1ad39dbdbede5064de2f7R20